### PR TITLE
chore: lint main with latest golangci-lint

### PR DESCRIPTION
**Pull request by faddat**
This PR is a regular update of the linter for wasmd

- bump golang and linter in ci
- cleanup lints

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ version: 2.1
 executors:
   golang:
     docker:
-      - image: cimg/go:1.19
```
Comment (alpe) [inline .circleci/config.yml:6]: We are still on [Go 1.19](https://github.com/CosmWasm/wasmd/blob/main/go.mod#L3) . Let's have the Go upgrade in a separate PR. We have https://github.com/CosmWasm/wasmd/issues/1385 for the next release

**Review by alpe — CHANGES_REQUESTED** [review]
Almost perfect! Please keep the Go version bump out of this PR. Happy to merge it then 💐

File: .circleci/config.yml (hunk 1 above)
Comment (faddat) [reply .circleci/config.yml:6]: sure thing

**Review by alpe — APPROVED** [review]
LGTM, Thanks for the help! 💐
